test/tests/functional/pbs_server_periodic_hook.py

```diff
@@ -277,3 +277,134 @@ def test_sp_with_queuejob(self):
         self.server.log_match(self.start_msg, interval=3)
         self.server.log_match(self.end_msg, interval=3)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_alarm_more_than_freq(self):
+        """
+        Test when alarm is more than freq. Ensure multiple
+        instances do not get launched
+        """
+        hook_name = "medium_hook"
+        scr = self.create_hook(accept=True, sleep_time=10)
+        attrs = {'event': 'periodic', 'alarm': 15, 'freq': 5}
+        self.server.create_import_hook(hook_name, attrs, scr, overwrite=True)
+        self.check_next_occurances(2, freq=5, hook_run_time=10,
+                                   check_for_hook_end=True)
+
+    def test_check_for_negative_freq(self):
+        """
+        Check for the correct messages thrown if negative values of freq is set
+        """
+        hook_name = "med_hook"
+        attrs = {'event': "periodic", 'freq': "0"}
+        match_str1 = "qmgr obj=med_hook svr=default:"
+        match_str1 += " set_hook_freq: freq value '0'"
+        match_str1 += " of a hook must be > 0"
+        match_str2 = "qmgr: hook error returned from server"
+        try:
+            self.server.create_hook(hook_name, attrs)
+        except PbsManagerError as e:
```

Comment:
The problem with this try/except is that if the exception does not get raised, the test still passes.  You have two options (here and the try/except further down): 1) use an else: with a self.acceptTrue(False, 'msg') or you can set a variable to False ahead of the try, set it to True in the except, and assertTrue() it after the try/except.

Reply:
@Mahalty you didn't fix the try/except.  It still has the same problem that the test will pass if an exception is not raised.  Since you are doing a negative test, the test should fail if the exception is not raised.